pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -187,7 +201,22 @@ func (i *RegistriesConf) Load(f asset.FileFetcher) (bool, error) {
 		return false, errors.Wrap(err, fmt.Sprintf("failed to load %s file", RegistriesConfFilename))
 	}
 
-	i.File = file
+	registriesConf := &sysregistriesv2.V2RegistriesConf{}
+	if err := toml.Unmarshal(file.Data, registriesConf); err != nil {
+		return false, errors.Wrapf(err, "failed to unmarshal %s", RegistriesConfFilename)
+	}
+
+	i.File, i.Config = file, registriesConf
+	i.setMirrorConfig(i.Config)
+
+	if string(i.File.Data) != defaultRegistriesConf {
```

Comment:
In which case this could happen? The `Load` is invoked when something was found on the disk, and by assuming it was provided by the user then validations should be applied always

Reply:
It could happen if the user ran `agent create cluster-manifests` and then didn't edit it, I think.
This check concerns me a bit, because we have had issues in the past where validating an empty registries.conf (which the default one is one example of) failed, and I suspect this check results in us not validating that any more in the unit tests.